src/test/java/seedu/intrack/model/internship/TaskTest.java

```diff
@@ -0,0 +1,32 @@
+package seedu.intrack.model.internship;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.intrack.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class TaskTest {
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Task(null, null));
+    }
+
+    @Test
+    public void constructor_invalidTask_throwsIllegalArgumentException() {
+        String invalidTaskName = "";
+        String invalidTaskTime = "TODAY";
+        assertThrows(IllegalArgumentException.class, () -> new Task(invalidTaskName, invalidTaskTime));
+    }
+
+    @Test
+    public void isValidTask() {
+        // null task
+        assertThrows(NullPointerException.class, () -> Task.isValidTask(null));
+
+        // valid tasks
+        assertTrue(Task.isValidTask(new Task("Online Assessment", "10-10-2023 10:00")));
+        assertTrue(Task.isValidTask(new Task("-", "05-11-2022 11:00"))); // one character
+        assertTrue(Task.isValidTask(new Task("The quick brown fox jumped over the lazy dog while testing"
+                + "the code in his project that is very very long", "20-10-2022 15:00"))); // long taskName
+    }
```

Comment:
Maybe you can consider splitting these into individual test cases for better coherency and ease in reading the test case file?